### PR TITLE
Bump to discv5 experimental branch for concurrent requests

### PR DESCRIPTION
**Pull request by njgheorghita**
### What was wrong?
Bump discv5 dependency to branch that contains support for concurrent requests.

### To-Do

[//]: # (Stay ahead of things, add list items here!)
- [ ] Add entry to the [release notes](https://github.com/ethereum/trin/blob/master/newsfragments/README.md) (may forgo for trivial changes)
- [ ] Clean up commit history

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -15,7 +15,7 @@ description = "A Rust implementation of the Ethereum Portal Network"
 [dependencies]
 anyhow = "1.0.68"
 clap = { version = "4.2.1", features = ["derive"] }
-discv5 = { version = "0.3.1", features = ["serde"] }
+discv5 = { git = "https://github.com/njgheorghita/discv5.git", rev = "6ab079e60f59b4124e024b1cee0f202efe3f0c87", features = ["serde"] }
```
Comment (carver) [inline Cargo.toml:18]: different revisions on different packages?

File: ethportal-api/Cargo.toml [hunk 2]
```diff
@@ -15,7 +15,7 @@ anyhow = "1.0.68"
 base64 = "0.13.0"
 bytes = "1.3.0"
 clap = { version = "4.2.1", features = ["derive"] }
-discv5 = { version = "0.3.1", features = ["serde"]}
+discv5 = { git = "https://github.com/njgheorghita/discv5.git", rev = "09262ead2882e062de1a6a45df9ea61917eb8465", features = ["serde"] }
```
Comment (carver) [inline ethportal-api/Cargo.toml:18]: other one starts 6ab

**Review by KolbyML — APPROVED** [review]
That is cool I didn't know you could pull by github commit.

**Comment by njgheorghita** [comment]
@KolbyML Yeah, you can also just do it off the branch name, eg: `branch = "experimental"`, but when debugging it's a little more peace of mind to know that you're building off of the exact commit version you want, rather than worrying about whether or not the latest commit to "experimental" was fetched correctly.... Although, typically you'll see the rev hash in the compile logs when building off a branch, but still, the rev method is foolproof imo.

**Review by carver — APPROVED** [review]
LGTM! Someday we'll be more cautious about using unmerged/unapproved changes to complex libraries like discv5. But today is not that day!

Reliability went way up according to glados, and it's easy enough to revert if we need to quickly. Eventually, we'll switch back to their released version anyhow.

Thanks for hunting this down! We were not going to get Portal live without parallel discv5 packets...

**Review by ogenev — APPROVED** [review]
:rocket:

**Comment by njgheorghita** [comment]
Big shoutout to @ackintosh who did most of the heavy lifting in squashing the final bugs, thanks!

**Pull request merged**